pyrenew_covid_wastewater/hosp_only_ww_model.py

```diff
@@ -0,0 +1,365 @@
+
+        inf_with_feedback_proc_sample = self.inf_with_feedback_proc(
+            Rt=rtu,
+            I0=i0,
+            gen_int=generation_interval_pmf,
+        )
+
+        latent_infections = jnp.concat(
+            [
+                i0,
+                inf_with_feedback_proc_sample.post_initialization_infections.value,
+            ]
+        )
+        numpyro.deterministic("rtu", rtu)
+        numpyro.deterministic("rt", inf_with_feedback_proc_sample.rt.value)
+        numpyro.deterministic("latent_infections", latent_infections)
+
+        p_hosp_mean = self.p_hosp_mean_rv()[0].value
+        p_hosp_w_sd = self.p_hosp_w_sd_rv()[0].value
+        autoreg_p_hosp = self.autoreg_p_hosp_rv()[0].value
+
+        p_hosp_ar_init_rv = DistributionalVariable(
+            "p_hosp_ar_init",
+            dist.Normal(
+                0,
+                p_hosp_w_sd / jnp.sqrt(1 - jnp.pow(autoreg_p_hosp, 2)),
+            ),
+        )
+        p_hosp_ar_init = p_hosp_ar_init_rv()[0].value
+
+        p_hosp_ar = self.p_hosp_ar_proc(
+            n=n_weeks_post_init,
+            autoreg=autoreg_p_hosp,
+            init_vals=p_hosp_ar_init,
+            noise_sd=p_hosp_w_sd,
+        )[0].value
+
+        ihr = jnp.repeat(
+            transformation.SigmoidTransform()(p_hosp_ar + p_hosp_mean),
+            repeats=7,
+        )[:n_datapoints]
+        # this is only applied after the hospital_admissions are generated, not to all the latent infections. This is why we cannot apply the ihr in compute_delay_ascertained_incidence
+        # see https://github.com/CDCgov/ww-inference-model/issues/43
+
+        numpyro.deterministic("ihr", ihr)
+
+        hosp_wday_effect_raw = self.hosp_wday_effect_rv()[0].value
+        hosp_wday_effect = tile_until_n(hosp_wday_effect_raw, n_datapoints)
+
+        inf_to_hosp = self.inf_to_hosp_rv()[0].value
+
+        potential_latent_hospital_admissions = (
+            compute_delay_ascertained_incidence(
+                p_observed_given_incident=1,
+                latent_incidence=latent_infections,
+                delay_incidence_to_observation_pmf=inf_to_hosp,
+            )[-n_datapoints:]
+        )
+
+        latent_hospital_admissions = (
+            potential_latent_hospital_admissions
+            * ihr
+            * hosp_wday_effect
+            * self.state_pop
+        )
+
+        hospital_admission_obs_rv = NegativeBinomialObservation(
+            "observed_hospital_admissions", concentration_rv=self.phi_rv
+        )
+
+        observed_hospital_admissions = hospital_admission_obs_rv(
+            mu=latent_hospital_admissions,
+            obs=data_observed_hospital_admissions,
+        )
+
+        return observed_hospital_admissions
+
+
+def create_hosp_only_ww_model_from_stan_data(stan_data_file):
+    with open(
+        stan_data_file,
+        "r",
+    ) as file:
+        stan_data = json.load(file)
+
+    def convert_to_logmean_log_sd(mean, sd):
```

Comment:
This function can be added to `plotting.py` with renaming of `plotting.py` or put in a separate file to share across model(s)